libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -191,6 +191,21 @@ Declaration const* NameAndTypeResolver::pathFromCurrentScope(vector<ASTString> c
 {
 	solAssert(!_path.empty(), "");
 	vector<Declaration const*> candidates = m_currentScope->resolveName(_path.front(), true);
+
+	if (!candidates.empty() && candidates.front()->isStructMember())
+	{
+		// A special case like
+		// enum E {a}; struct S {E E;}
+		// We are within a struct scope and resolved name is a struct member name.
```

Comment:
Maybe we can add an assert here about being inside struct scope?

Reply:
I did not find an easy way to check if we are within a struct scope (and that impacted my previous implementation).

I extended `DeclarationContainer` to provide that check, see https://github.com/ethereum/solidity/pull/10908/commits/085bede098e1c5123c778283ea1836343aba705f. Is it worth it?